src/components/org/OrgChart/hooks/useParentSize.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react';
+
+type Size = {
+    height: number;
+    width: number;
+};
+
+export default (ref?: React.MutableRefObject<SVGElement | null>): [number, number] => {
+    const [size, setSize] = useState<Size>({ height: 0, width: 0 });
+
+    const handleResize = () => {
+        if (ref && ref.current && ref.current.parentElement) {
+            const parent = ref.current.parentElement;
+            setSize({ height: parent.offsetHeight, width: parent.offsetWidth });
+        }
+    };
+    useEffect(() => {
```

Comment:
Here we could use useEventListener(window, 'resize', handleResize); or just a loop with requestAnimationFrame. The more event listeners we have on resize (especially with layout thrashing) the jankier the application gets

Reply:
Now uses requestAnimationFrame